src/linter/block_linter.go

```diff
@@ -184,6 +184,39 @@ func (b *blockLinter) enterNode(n ir.Node) {
 	}
 }
 
+func (b *blockLinter) checkForeach(n *ir.ForeachStmt) {
+	b.walker.r.checkKeywordCase(n, "foreach")
+
+	var vars []*ir.SimpleVar
+
+	findAllVars := func(n ir.Node) bool {
+		if vr, ok := n.(*ir.SimpleVar); ok {
+			vars = append(vars, vr)
+		}
+		return true
+	}
+
+	if n.Variable != nil {
+		irutil.Inspect(n.Variable, findAllVars)
+	}
+	if n.Key != nil {
+		irutil.Inspect(n.Key, findAllVars)
+	}
+
+	fun, ok := b.walker.r.currentFunction()
+	if !ok {
+		return
+	}
+
+	for _, v := range vars {
+		for _, param := range fun.Params {
```

Comment:
I'm OK with a nested loop here as it will never be a bottleneck, probably.
But we have functions with 1k+ lines of code, many foreach loops (and probably many params).

How about creating a map inside a walker when we enter a function that would hold all params?
It should probably be lazily initialized, the first time we call `getCurrentFuncParams()` method.